### ISSUE TITLE
vaft.formula.stability -> utils -> sklearn costs ~0.8 s at import

### ISSUE THREAD

**Issue by HongSik-Yun-Fusion** [VEST-Tokamak/vaft#367]
```
python -X importtime -c "import vaft.formula.stability"
```

shows a self-time of well under a millisecond against a cumulative cost measured in seconds. Re-measured 2026-09-06:

```
                              self        cumulative
sklearn                     1 577 us     1 554 289 us
sklearn.gaussian_process       694 us     1 726 906 us
vaft.formula.utils             398 us     1 727 303 us
vaft.formula.stability         554 us     2 570 100 us
```

The absolute numbers are machine-dependent and have grown since this issue was filed (~0.9 s then, ~1.7 s for sklearn alone now), but the shape is the point: **0.55 ms of `stability` costs 2.57 s to import.** Almost all of it comes from `vaft.formula.utils`, which imports sklearn at module level:

```python
from scipy.optimize import curve_fit
from sklearn.gaussian_process import GaussianProcessRegressor
from sklearn.gaussian_process.kernels import RBF, ConstantKernel as C
```

Those exist for the single `fit_profile(fitting_function='gp')` branch. Every physics submodule that needs `utils.gradient` — which is most of them — pays the full sklearn import for a Gaussian process it will never construct.

## Why the existing perf test does not catch this

`test/test_formula_catalog.py::test_the_normal_import_path_stays_cheap` looks like it should, and does not, for two independent reasons:

1. **It measures self-time, not cumulative.** The regex captures `match.group(1)`, the first `-X importtime` column. `vaft.formula.stability`'s self-time is ~0.2 ms and passes the 50 ms budget comfortably while its cumulative cost is 0.9 s.
2. **`vaft.formula.utils` is not in the checked list.** The loop covers only `vaft.formula`, `vaft.formula.constants` and `vaft.formula.stability` — the module that actually pulls sklearn is never asserted on.

Both have to change. Adding `utils` to the list while still reading self-time would still pass.

## Suggested fix

Move `GaussianProcessRegressor`, `RBF`, `ConstantKernel` and `curve_fit` inside `fit_profile`, and extend the perf test to assert a **cumulative** budget for `vaft.formula.stability` (the `match.group(2)` column), so the regression this issue describes cannot come back unnoticed.

## Blast radius

`test/test_formula_catalog.py::test_the_catalog_counts_the_known_public_surface` asserts exact per-category function counts (`utils: 9`). Moving imports does not change them, but any accompanying refactor of `utils.py` that adds or renames a public function does, and the count must be updated in the same commit.

---
Found during the issue #248 documentation pass. The behaviour is now stated under *Limitations* in the function docstring (`vaft.formula.describe(...)`) and left unchanged there; this issue tracks the fix.

**Comment by HongSik-Yun-Fusion** [comment]
Closing as a duplicate of #426, which tracks the same defect with wider scope.

**Same defect.** Both describe `vaft/formula/utils.py` importing scikit-learn at module scope
(`:11-12`) for `fit_profile`'s `'gp'` branch alone, so every module that needs `utils.gradient` pays
the full sklearn import.

**Why #426 is the survivor**, despite being filed a day later (2026-09-03 vs 2026-09-02):

- it carries acceptance criteria;
- it names the downstream blast radius — `vaft/process/profile.py` does `from vaft.formula import
  fit_profile`, so `import vaft.process.profile` costs 1604 modules against the 837 of `import vaft`;
- it names the `vaft/formula/green.py:41` workaround that exists *only* because of this import, and
  which can be retired with it;
- it sits with the #249 import audit it came from, which is where the rest of this class of work is
  tracked.

**Nothing is lost.** #426 now carries this issue's two unique findings — the analysis of why
`test_the_normal_import_path_stays_cheap` never caught this (it reads the self-time column,
`match.group(1)`, and omits `vaft.formula.utils` from its checked list, so fixing either alone still
passes) and the 2026-09-06 re-measurement (`vaft.formula.stability`: 0.55 ms self, 2.57 s cumulative).

**The scope also grew.** #426 is no longer "move the import" but "make scikit-learn optional": add a
scipy-native GP, keep sklearn behind an opt-in, drop `scikit-learn` from `[project] dependencies`,
and add a test comparing the two implementations. sklearn's whole footprint is two sites, and the
second (`vaft/process/statistical_analysis.py:492`, `r2_score` / `mean_squared_error`) is a few lines
of numpy.

Verified still present on `develop` @ `86c7a24`: `from sklearn.gaussian_process import` is unchanged
in `vaft/formula/utils.py`, and `scikit-learn` remains a hard runtime dependency at
`pyproject.toml:52`.

Milestone 22 goes 20 → 19 open. Work continues in #426.
